tests/galsim_test_helpers.py

```diff
@@ -204,22 +204,10 @@ def do_kvalue(prof, name):
             err_msg = name +
             " convolved with a delta function is inconsistent with real-space image.")
 
-
-def radial_integrate(prof, minr, maxr, dr):
-    """A simple helper that calculates int 2pi r f(r) dr, from rmin to rmax
-       for an axially symmetric profile.
-    """
-    import math
+def radial_integrate(prof, minr, maxr):
     assert prof.isAxisymmetric()
-    r = minr
-    sum = 0.
-    while r < maxr:
-        # In this tight loop, it is worth optimizing away the parse_pos_args step.
-        # It makes a rather significant difference in the running time of this function.
-        sum += r * prof.SBProfile.xValue(galsim.PositionD(r,0)) 
-        r += dr
-    sum *= 2. * math.pi * dr
-    return sum
+    f = lambda r: 2 * np.pi * r * prof.SBProfile.xValue(galsim.PositionD(r,0))
```

Comment:
Similarly, it is worth keeping my comment about why we use `prof.SBProfile.xValue` rather than `prof.xValue`.  In the latter case, the argument checking of the python version slows the function evaluation considerably, which matters when the xValues are being called in a tight loop.  (Both in the original version, and in your improved int1d version.)